### PR TITLE
Direct Hits from Tesla Ball no longer drains from the Plasmadrain Immune

### PR DESCRIPTION
**Pull request by Runian**
## About The Pull Request
Direct hits from a tesla no longer drains plasma from castes that have the CASTE_PLASMADRAIN_IMMUNE caste flag.

## Why It's Good For The Game
The castes are immune to plasma drain. This tesla ball drains plasma from castes that are immune when they aren't suppose to. Bugfix.

## Changelog
:cl:
fix: Xeno castes who are suppose to be immune from plasma drain no longer have their plasma drained when directly hit by a tesla ball.
/:cl:

### CODE REVIEW

**Comment by Zangeiti** [comment]
an incredibly niche tool that drains plasma much slower than the pepperball (which does what the tesla ball does but 10x better)
keep the telsa's "bug" so it serves a purpose other than being a direct downgrade to pepperball

**Comment by SimplySmiley123** [comment]
I rather see this turned into a feature then "fixed".
Tesla is horrid enough at it's job as is.

**Comment by Runian** [comment]
> an incredibly niche tool that drains plasma much slower than the pepperball (which does what the tesla ball does but 10x better) keep the telsa's "bug" so it serves a purpose other than being a direct downgrade to pepperball

It already serves a purpose because it does something that pepperball doesn't. It slows down nearby xenos, can be IFF, can be used with BYOS, and doesn't require lot of aiming besides firing in a general direction.

**Comment by Zangeiti** [comment]
> It already serves a purpose because it does something that pepperball doesn't. It slows down nearby xenos, can be IFF, can be used with BYOS, and doesn't require lot of aiming besides firing in a general direction.

Slowdown is negligible given how short it lasts, "IFF" but still can hit marines if direct hit, Noone ever uses it with BYOS, and the pepperball has 100 pellets to spray with an incredibly high firerate vs. 6 tesla balls per mag that fires once every !!FOUR!! seconds

File: code/modules/projectiles/ammo_types/energy_ammo.dm [hunk 1]
```diff
@@ -70,7 +70,8 @@
 /datum/ammo/energy/tesla/on_hit_mob(mob/target_mob, obj/projectile/proj)
 	if(isxeno(target_mob)) //need 1 second more than the actual effect time
 		var/mob/living/carbon/xenomorph/X = target_mob
-		X.use_plasma(0.3 * X.xeno_caste.plasma_max * X.xeno_caste.plasma_regen_limit) //Drains 30% of max plasma on hit
+		if(!(X.xeno_caste.caste_flags & CASTE_PLASMADRAIN_IMMUNE))
```
Comment (Lumipharon) [inline code/modules/projectiles/ammo_types/energy_ammo.dm:73]: Early return

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**